### PR TITLE
fix: better output colors

### PR DESCRIPTION
**Pull request by wraithgar**
No more grey, no more setting background color.

Added the beginnings of guidance to the display layer comments.

### CODE REVIEW

**Comment by wraithgar** [comment]
Maybe we squash this into `fix: better output colors`

File: lib/commands/audit.js [hunk 1]
```diff
@@ -104,7 +104,7 @@ class VerifySignatures {
     }
 
     if (missing.length) {
-      const missingClr = this.npm.chalk.bold(this.npm.chalk.red('missing'))
+      const missingClr = this.npm.chalk.red.bold('missing')
```
Comment (wraithgar) [inline lib/commands/audit.js]: redBright

File: lib/commands/audit.js [hunk 2]
```diff
@@ -122,7 +122,7 @@ class VerifySignatures {
       if (missing.length) {
         this.npm.output('')
       }
-      const invalidClr = this.npm.chalk.bold(this.npm.chalk.red('invalid'))
+      const invalidClr = this.npm.chalk.red.bold('invalid')
```
Comment (wraithgar) [inline lib/commands/audit.js]: redBright

File: lib/commands/fund.js [hunk 3]
```diff
@@ -153,7 +152,8 @@ class Fund extends ArboristWorkspaceCmd {
     })
 
     const res = tree(result)
-    return this.npm.chalk.reset(res)
+    return res
+    // return this.npm.chalk.reset(res)
```
Comment (wraithgar) [inline lib/commands/fund.js]: Chesterton's fence. Why was this here?

File: lib/commands/view.js [hunk 4]
```diff
@@ -318,125 +318,89 @@ class View extends BaseCommand {
...
+      this.npm.output(chalk.blue(site))
     }
 
-    const warningSign = unicode ? ' ⚠️ ' : '!!'
-    info.deprecated && this.npm.output(
-      `\n${chalk.bold.red('DEPRECATED')}${
-      warningSign
-    } - ${info.deprecated}`
+    manifest.deprecated && this.npm.output(
+      `\n${chalk.bold.red('DEPRECATED')}${unicode ? ' ⚠️ ' : '!!'} - ${manifest.deprecated}`
```
Comment (wraithgar) [inline lib/commands/view.js]: redBright

File: lib/utils/display.js [hunk 5]
```diff
@@ -3,18 +3,32 @@ const log = require('proc-log')
...
+  White # (Don't use)
+*/
+
+// Translates log levels to chalk colors
 const COLOR_PALETTE = ({ chalk: c }) => ({
-  heading: c.white.bgBlack,
-  title: c.magenta,
-  timing: c.green.bgBlack,
+  heading: c.bold,
+  title: c.blue.bold,
```
Comment (wraithgar) [inline lib/utils/display.js]: blueBright?  Applies to all bolds in this palette.

File: lib/commands/view.js [hunk 6]
```diff
@@ -318,125 +318,89 @@ class View extends BaseCommand {
...
-    if (info.deps.length) {
-      this.npm.output('')
-      this.npm.output('dependencies:')
-      this.npm.output(columns(info.deps.slice(0, maxDeps), { padding: 1 }))
-      if (info.deps.length > maxDeps) {
-        this.npm.output(`(...and ${info.deps.length - maxDeps} more.)`)
+    this.npm.output('\ndist')
+    this.npm.output(`.tarball: ${chalk.blue(manifest.dist.tarball)}`)
+    this.npm.output(`.shasum: ${chalk.green(manifest.dist.shasum)}`)
+    if (manifest.dist.integrity) {
```
Comment (wraithgar) [inline lib/commands/view.js:363]: The `else` of this isn't covered in tests, even though before the code was

```
manifest.dist.integrity && chalk.yellow(manifest.dist.integrity)
```

**Review by lukekarrys — APPROVED** [review]

**Comment by wraithgar** [comment]
@lukekarrys rebased and fixed merge conflicts in the view command.

**Review by lukekarrys — APPROVED** [review]

File: lib/utils/display.js [hunk 7]
```diff
@@ -3,18 +3,33 @@ const { log, output } = require('proc-log')
...
+/*
+ * Black # (Don't use)
+ * Red # Danger
+ * Green # Success
+ * Yellow # Warning
+ * Blue # Accent
+ * Magenta # Done
+ * Cyan # Emphasis
+ * White # (Don't use)
+ */
```
Comment (andyfeller) [inline lib/utils/display.js:18]: @wraithgar : could you elaborate on why black and white are not explicitly used?

I assume it's because we'll defer to the terminal's default coloring which should hopefully allow users to colorize as desired.

Comment (wraithgar) [reply lib/utils/display.js:18]: > I assume it's because we'll defer to the terminal's default coloring which should hopefully allow users to colorize as desired.

It's exactly that.

Typically a terminal will configure a separate foreground and background color, discrete from the customizable 16 color palette. From what I've seen they will not also adjust black and white in the palette to be in accordance w/ the foreground/background colors.

If one wants to display with the foreground color, then print with "no" coloring.  There is also bold and dim if you need those.  In cases where you really do want the background color, you can inverse (I'd probably still avoid that though).
